housekeeping

iamjohnnym opened the pull request:
There was an issue with `pip.req` in newer versions of pip that would prevent `setup.py` to function as expected.
I also did some general house keeping on package versions in the requirements.py

* Added `bandit` for vulnerability testing
* removed debug in the app.run()
* newest version of pip deprecated pip.req
* Added bash shabang
* added more versions to test with travis
* travis integration

[![Build Status](https://travis-ci.org/iamjohnnym/flask-boto3.svg?branch=master)](https://travis-ci.org/iamjohnnym/flask-boto3)

iamjohnnym committed:
Travis integration (#1)

* travis integration

* fixed filename

* fix test packages

* fixed bandit

* cfixed error coming from bandit

iamjohnnym committed:
bumped max version # of flask and boto3 to include latest

* bumped versions of flask and boto3 to include a higher max version 0.12.2 and 1.7.4

iamjohnnym committed:
housekeeping

* removed debug in the app.run()
* newest version of pip deprecated pip.req
* Added bash shabang
* added more versions to test with travis
* sneaky word has been found

Commit:
added bandit to circle

iamjohnnym commented:
After looking into it, it appears circle is now out of date.  I'll look into getting that squared away.

Ketouem commented:
Awesome job that you did @iamjohnnym, many thanks !

Several remarks though:
- I manage all my project using CircleCI and I'd like not to use multiple CI services, so for testing multiple versions of Python I'd go towards using `tox` inside a CircleCI build
- Don't bother with changing the version number, I'll do it.

iamjohnnym commented:
Absolutely.  I should have some time tonight to go through that.  I'm not familiar with circle so I defaulted to travis, for that I apologize for imposing that.  

For continued consistency, what do you set as `$CIRCLE_TEST_REPORTS` and `$CIRCLE_ARTIFACTS`?  I'd like to replicate that myself.

Ketouem commented:
Actually the two variables that you mentioned are already set by CircleCI during the builds. But it's still up to you to output files to this location as per demonstrated in the examples in the documentation https://circleci.com/docs/1.0/test-metadata/.

Btw it should be a good idea to migrate to 2.0 but that is something I can do afterwards if you're not comfy with.

iamjohnnym committed:
CircleCI v2 with tox

- Added Tox Support
- Check py27,py34,py35,py36
- CircleCI v2 config

iamjohnnym commented:
I got started on the `circleci` v2 with `Tox`.  It was a bit more of a challenge than I anticipated.  It can now test, py27,py34,py35,py36.  

I changed the image to `themattrix/tox` which seems to be a goto, maybe?, for tox builds.  However, you lose all visibility on the output.  It may not have been the best choice but the last few hours were my only exposure to it `tox` and `circleci`.

The only missing piece is the `deployment` aspect.

Ketouem commented:
Something like 
```yaml
python: &python
  docker:
    - image: circleci/python:3.6
      environment: ~/.bashrc

version: 2

jobs:
  build:
    <<: *python
    steps:
      - checkout
      - run:
          name: Update PATH
          command: echo 'export PATH=$HOME/.local/bin:$PATH' >> $BASH_ENV
      - run:
          name: Install tox
          command: pip install --user $(cat requirements_test.txt | grep tox)
      - run:
          name: Run tests
          command: TOXBUILD=true tox
      - store_artifacts:
          path: test-reports
  deploy:
    <<: *python
    steps:
      - checkout
      - run:
          name: Update PATH
          command: echo 'export PATH=$HOME/.local/bin:$PATH' >> $BASH_ENV
      - run:
          name: Install Twine
          command: pip install --user twine
      - run:
          name: Push to cheeseshop
          command: ./push_to_cheeseshop.sh

workflows:
  version: 2
  build-test-deploy:
    jobs:
      - build:
          filters:
            tags:
              only: /.*/
      - deploy:
          requires:
            - build
          filters:
            tags:
              only: /^\d+(?:\.\d+){2}$/
            branches:
              only: master
```
should do the trick

Ketouem commented on .gitignore:
```diff
@@ -83,3 +88,6 @@ ENV/
 # Spyder project settings
 .spyderproject
+
+# Travis
```
No longer needed as we're not using Travis

Ketouem commented on Makefile:
```diff
@@ -0,0 +1,21 @@
+
+.PHONY: circle_test
+circle_test:
+	nosetests --with-xunit --xunit-file=$CIRCLE_TEST_REPORTS/xunit.xml --cover-branches --with-coverage --cover-erase --cover-package=flask_boto3 --cover-html --cover-html-dir=$CIRCLE_ARTIFACTS/coverage
```
`CIRCLE_TEST_REPORTS` & `CIRCLE_ARTIFACTS` are used with Circle 1.0. Collecting test data is now performed with specific steps cf https://circleci.com/docs/2.0/collect-test-data/ you may need to create output folder beforehand and reference them in the CircleCI configuration.

Ketouem commented on .circleci/config.yml:
```diff
@@ -0,0 +1,34 @@
+# Python CircleCI 2.0 configuration file
```
Something like should do the trick for the deployment
```
python: &python
  docker:
    - image: circleci/python:3.6
      environment: ~/.bashrc

version: 2

jobs:
  build:
    <<: *python
    steps:
      - checkout
      - run:
          name: Update PATH
          command: echo 'export PATH=$HOME/.local/bin:$PATH' >> $BASH_ENV
      - run:
          name: Install tox
          command: pip install --user $(cat requirements_test.txt | grep tox)
      - run:
          name: Run tests
          command: TOXBUILD=true tox
      - store_artifacts:
          path: test-reports
  deploy:
    <<: *python
    steps:
      - checkout
      - run:
          name: Update PATH
          command: echo 'export PATH=$HOME/.local/bin:$PATH' >> $BASH_ENV
      - run:
          name: Install Twine
          command: pip install --user twine
      - run:
          name: Push to cheeseshop
          command: ./push_to_cheeseshop.sh

workflows:
  version: 2
  build-test-deploy:
    jobs:
      - build:
          filters:
            tags:
              only: /.*/
      - deploy:
          requires:
            - build
          filters:
            tags:
              only: /^\d+(?:\.\d+){2}$/
            branches:
              only: master
```

Ketouem commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+FROM themattrix/tox-base
```
Base Python image can be used to install all what is needed, see comments about `config.yml`.

iamjohnnym committed:
Pr3 change requests

- Migrated to Circleci v2
- Remove unnecessary info in .gitignore
- Changed paths for tests and artifacts
- Tests include 2.7 and 3.6 via `tox`
- Fixed setup.py for `pip>=10.0.0`
